### PR TITLE
cpu_clock_init() implementation

### PR DESCRIPTION
**Pull request by fvcoen**
Support for selecting the sources of `SCLK_HF` and `SCLK_LF`.

### CODE REVIEW

**Comment by yogo1212** [comment]
the commits concerning UNIFLASH_PATH and .gitignore aren't necessary for the clock-initialisation.
i'm afraid adding Makefile.include to .gitignore isn't an option :-(
i use this alias: `alias cc2650stkmake='UNIFLASH_PATH=/home/ti/ti/ccsv6 BOARD=cc2650stk make'`
maybe you want to work with stashes locally instead?

File: cpu/cc26x0/include/cpu_conf.h [hunk 1]
```diff
@@ -49,6 +49,24 @@ extern "C" {
...
+
+/* configure HF clock at init by setting 1 for the desired source */
+/* HF clock is used by MCU_VD in active+idle modes and AUX_PD in standby mode */
+#define HF_CLOCK_HF_ROSC 1  /* set 48MHz RCOSC as HF system clock */
+#define HF_CLOCK_HF_XOSC 0  /* set 48MHz XOSC (24MHz doubled internally) as HF system clock */
+
+/* configure LF clock at init by setting 1 for the desired source */
+/* LF clock is available to MCU_VD in standby modes, available to AUX_PD in active mode and is always used by AON */
+#define LF_CLOCK_HF_ROSC  1 /* set 31.25kHz derived from 48MHz RCOSC as LF system clock */
+#define LF_CLOCK_HF_XOSC  0 /* set 31.25kHz derived from 24MHz XOSC as LF system clock */
```
Comment (yogo1212) [inline cpu/cc26x0/include/cpu_conf.h]: even though their value is 0 they are defined nonetheless -> `#elif LF_CLOCK_HF_XOSC` would still be taken ;-)

File: cpu/cc26x0/cpu.c [hunk 2]
```diff
@@ -17,13 +17,47 @@
...
+   AON_WUC->AUXCTL |= AUXCTL_AUX_FORCE_ON; /* power on AUX_PD */
+   while(!(AON_WUC->PWRSTAT & PWRSTAT_AUX_PD_ON)); /* wait for AUX_PD to be powered on */
+   AUX_WUC->MODCLKEN0 |= MODCLKEN0_AUX_DDI0_OSC_EN; /*turn on oscillator interface clock */
+
+   /* configure HF clock */
+   #if HF_CLOCK_HF_ROSC
+     DDI_0_OSC->CTL0 |= 0x0; /* set 48MHz RCOSC */
+   #elif HF_CLOCK_HF_XOSC
+     DDI_0_OSC->CTL0 |= DDI_0_OSC_CTL0_SCLK_HF_SRC_SEL; /* set 48MHz XOSC (24MHz doubled internally) */
+   #endif
```
Comment (yogo1212) [inline cpu/cc26x0/cpu.c]: the preprocessor branching is a bit ugly :-)
`DDI_0_OSC->CTL0 |= 0x0;` should be enough for the start.
we can think about making that configurable later.

File: cpu/cc26x0/cpu.c [hunk 3]
```diff
@@ -17,13 +17,47 @@
...
+   #if LF_CLOCK_HF_RCOSC
+     DDI_0_OSC->CTL0 |= DDI_0_OSC_CTL0_SCLK_LF_SRC_SEL_HF_RCOSC; /* set 31.25kHz derived from 48MHz RCOSC */
+   #elif LF_CLOCK_HF_XOSC
+     DDI_0_OSC->CTL0 |= DDI_0_OSC_CTL0_SCLK_LF_SRC_SEL_HF_XOSC; /* set 31.25kHz derived from 24MHz XOSC */
+   #elif LF_CLOCK_LF_RCOSC
+     DDI_0_OSC->CTL0 |= DDI_0_OSC_CTL0_SCLK_LF_SRC_SEL_LF_RCOSC; /* set 32kHz derived from 32kHz RCOSC */
+   #elif LF_CLOCK_LF_XOSC
+     DDI_0_OSC->CTL0 |= DDI_0_OSC_CTL0_SCLK_LF_SRC_SEL_LF_XOSC; /* set 32.768kHz derived from 32.768kHz XOSC */
+   #endif
+ }
```
Comment (yogo1212) [inline cpu/cc26x0/cpu.c:54]: same here.

these are many lines of code :-/
in the end, it should probably look more like this:

```
#ifndef HF_CLOCK_SOURCE
#define HF_CLOCK_SOURCE DDI_0_OSC_CTL0_SCLK_HF_SRC_SEL_RCOSC
#endif
#ifndef LF_CLOCK_SOURCE
#define LF_CLOCK_SOURCE DDI_0_OSC_CTL0_ACLK_REF_SRC_SEL_RCOSC_HF
#endif
...
static void cpu_clock_init(void)
...
DDI_0_OSC->CTL0 |= HF_CLOCK_SOURCE | LF_CLOCK_SOURCE;
```

**Comment by fvcoen** [comment]
Thank you for the feedback :)

> the commits concerning UNIFLASH_PATH and .gitignore aren't necessary for the clock-initialisation.
> i'm afraid adding Makefile.include to .gitignore isn't an option :-(

Sorry for the clumsy commits: I wanted to avoid submitting the `Makefile.include` file with my specific path but I'm still getting used to `git`. Thanks for suggesting using an `alias`, it's a much better solution!

I've adopted your approach for the rest of the code, it definitely improves readability :)

**Comment by yogo1212** [comment]
@fvcoen nice :-)
can you squash all the commits into one?

just in case:

```
# update cc2650stk_pr
# then
$ git checkout cc2650stk
$ git rebase -i cc2650stk_pr
# change the file's content to: (then save and quit)
p 5238f5f
f 6068698

# force push
```

**Comment by fvcoen** [comment]
@yogo1212 Done! Thanks for the git commands, it saved me a visit on stackoverflow ;)

**Pull request merged**